packages/core/src/components/dialog/_dialog.scss

```diff
@@ -130,6 +136,12 @@ $dialog-header-padding: math.div($pt-grid-size, 2);
     .#{$ns}-icon-large,
     .#{$ns}-icon {
       color: $pt-dark-icon-color;
+
+      @each $intent, $color in $pt-dark-intent-text-colors {
```

Comment:
Tried out both `$pt-intent-colors` and `$pt-dark-intent-text-colors` here for dark mode. The `$pt-intent-colors` that we use for light mode icon intents looked a bit too low contrast, so opted for the dark "text" intent colors instead. The only weird thing is using something name "intent-text-colors" for icons intents.

<img width="400" alt="pt-intent-colors" src="https://github.com/user-attachments/assets/1a0b5c9e-ccc2-4acc-9627-bd3ad7e55141">

<img width="400" alt="pt-dark-intent-text-colors" src="https://github.com/user-attachments/assets/8fe07129-180e-48b0-aa3e-afb9c35da997">